tern/analyze/default/container/image.py

```diff
@@ -73,3 +77,25 @@ def analyze(image_obj, options):
         passthrough.run_extension(image_obj, options.extend, options.redo)
     else:
         default_analyze(image_obj, options)
+
+
+def common_container_procedure(image, options, image_file=None):
+    """Setup the image object and anything else for analysis"""
+    # Add a Notice object for each layer
+    for layer in image.layers:
+        origin_str = 'Layer {}'.format(layer.layer_index)
+        layer.origins.add_notice_origin(origin_str)
+    # Set up working directories and mount points
+    rootfs.set_up()
+    # analyze the image
+    analyze(image, options)
+    if image_file == "Full_Image":
+        lock.set_imported_layers(image)
+    elif image_file == "Base_Image":
+        pass
+    else:
+        report.report_out(options, image)
```

Comment:
small nit: Can you remove the the check for base image since we don't do anything with it? i.e.

```
    if image_file == "Full_Image":
        lock.set_imported_layers(image)
    elif image_file is None:
        report.report_out(options, image)
```

Reply:
Surely, one more thing that I think I should change is `image_file` with `image_type`? Will it be better ?

Reply:
I think `image_file` is fine.